src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/helpers/model/SwitchPairs.groovy

```diff
@@ -57,14 +65,52 @@ class SwitchPairs {
     }
 
     SwitchPairs includeSwitch(Switch sw) {
-        return new SwitchPairs(switchPairs.findAll { it.src == sw || it.dst == sw})
+        return new SwitchPairs(switchPairs.findAll { it.src == sw || it.dst == sw })
     }
 
     SwitchPairs excludeSwitches(List<Switch> switchesList) {
-        return new SwitchPairs(switchPairs.findAll { !(it.src in switchesList) || !(it.dst in switchesList)})
+        return new SwitchPairs(switchPairs.findAll { !(it.src in switchesList) || !(it.dst in switchesList) })
     }
 
     List<Switch> collectSwitches() {
         switchPairs.collectMany { return [it.src, it.dst] }.unique()
     }
+
+    SwitchPairs withAtLeastNTraffgensOnSource(int traffgensConnectedToSource) {
+        return new SwitchPairs(
+                switchPairs.findAll { it.getSrc().getTraffGens().size() >= traffgensConnectedToSource })
+    }
+
+    SwitchPairs withBothSwitchesVxLanEnabled(SwitchHelper switchHelper) {
```

Comment:
WDYT to use switches props as a param or maybe a list of switches that support vxlan?